cls/SourceControl/Git/DiscardState.cls

```diff
@@ -0,0 +1,138 @@
+Class SourceControl.Git.DiscardState Extends (%Persistent, %JSON.Adaptor)
+{
+
+Property FullExternalName As %String(MAXLEN = "") [ Required ];
+
+Property Name As %String [ Required ];
+
+Property Contents As %Stream.GlobalCharacter(LOCATION = "^DiscardStateContents");
+
+Property Username As %String [ Required ];
+
+Property Branch As %String [ Required ];
+
+Property Timestamp As %TimeStamp [ Required ];
+
+Property ExternalFile As %Boolean [ Required ];
+
+Method RestoreToFileTree()
+{
+    // Make sure directory for file exists
+    set dir = ##class(%File).GetDirectory(..FullExternalName)
+    if ('##class(%File).DirectoryExists(dir)) {
+        do ##class(%File).CreateDirectoryChain(dir)
+    }
+    
+    // Recreate File
+    set fileStream = ##class(%Stream.FileCharacter).%New()
+    set fileStream.Filename = ..FullExternalName
+    $$$ThrowOnError(fileStream.CopyFrom(..Contents))
+    $$$ThrowOnError(fileStream.%Save())
+
+    // Add file to source-control / IRIS
+    if '..ExternalFile {
+        do ##class(SourceControl.Git.Utils).ImportItem(..Name, 1, 1)
+        do ##class(SourceControl.Git.Utils).AddToServerSideSourceControl(..Name)
+    }
+
+    // Delete discard record
+    $$$ThrowOnError(..%DeleteId(..%Id()))
+}
+
+ClassMethod SaveDiscardState(InternalName As %String, name As %String) As %Status
+{
+        set discardState = ..%New()
+        
+        if (InternalName = "") {
+            // If not in IRIS
+            set externalName = ##class(%File).Construct($Get(^SYS("SourceControl","Git","settings","namespaceTemp")),name)
```

Comment:
Can we replace the direct global reference with a method call to `SourceControl.Git.Utils`?